### PR TITLE
Misc fixes

### PR DESCRIPTION
**Pull request by someaddons**
# Changes proposed in this pull request:
- Fix guards being hit immune when not following or rallied
- Adjust pathing to prefer going down than going up, while also increasing the cost for dropping down multiple blocks
- Add fast client updates for citizens that are being interacted with by a player

[ x] Yes I tested this before submitting it.
[ ] I also did a multiplayer test.

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/core/entity/pathfinding/pathjobs/AbstractPathJob.java [hunk 1]
```diff
@@ -551,7 +551,7 @@ protected double computeCost(
             }
             else if (pathingOptions.dropCost != 1)
             {
-                cost *= pathingOptions.dropCost * Math.abs(dPos.getY());
+                cost *= pathingOptions.dropCost * Math.abs(dPos.getY() * dPos.getY());
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/core/entity/pathfinding/pathjobs/AbstractPathJob.java]: could we except stair blocks from this?

**Comment by uecasm** [comment]
> Adjust pathing to prefer going down than going up

I can see the logic for that with ordinary colonies, but I wonder if that might be counterproductive in cavern or other underground colonies.  Perhaps they should favour going toward y=64 when in doubt?  Or even y=townhall.

**Comment by Raycoms** [comment]
> > Adjust pathing to prefer going down than going up
> 
> I can see the logic for that with ordinary colonies, but I wonder if that might be counterproductive in cavern or other underground colonies. Perhaps they should favour going toward y=64 when in doubt? Or even y=townhall.

I understood it in a way that was: Fix pathing that was preferring this

**Comment by uecasm** [comment]
If I'm reading the code correctly, the cost of jumping up one block has changed from 1.1 to 1.5, the cost of dropping down one block has remained at 1.1, and the cost of dropping down two blocks has changed from 2.2 to 4.4.  (Though the last assumes that dPos.y can be -2, which the javadoc suggests should not happen, though perhaps it can.)

So that's _probably_ overall preferring down more than up, though only when not dropping more than one block.  (Also, walking up stairs might still count as jumping.)

File: src/main/java/com/minecolonies/core/entity/pathfinding/pathjobs/AbstractPathJob.java (hunk 1 above)
Comment (someaddons) [reply src/main/java/com/minecolonies/core/entity/pathfinding/pathjobs/AbstractPathJob.java]: stair blocks are always max 1 diff, this only affects jumping multiple blocks

**Comment by someaddons** [comment]
closing and reopening since bunch of stuff changed

**Pull request closed without merging**